tests/zero_code_change/test_tensorflow_disengage_smdebug.py

```diff
@@ -0,0 +1,66 @@
+# Third Party
+# Standard Library
+import os
+from unittest.mock import patch
+
+import pytest
+import tensorflow.compat.v2 as tf
+from tests.utils import SagemakerSimulator
+
+# First Party
+import smdebug.tensorflow as smd
+from smdebug.core.config_validator import reset_config_validator
+
+SMDEBUG_PREFIX = "smdebug_"
+
+
+@pytest.fixture(autouse=True)
+def cleanup():
+    yield
+    os.environ.pop("USE_SMDEBUG", None)
+    os.environ.pop("SM_HPS", None)
+    reset_config_validator()
+
+
+def get_keras_model_v2():
+    model = tf.keras.models.Sequential(
+        [
+            tf.keras.layers.Flatten(input_shape=(28, 28)),
+            tf.keras.layers.Dense(128, activation="relu"),
+            tf.keras.layers.Dropout(0.2),
+            tf.keras.layers.Dense(10, activation="softmax"),
+        ]
+    )
+    return model
+
+
+def get_keras_data():
+    mnist = tf.keras.datasets.mnist
+    (x_train, y_train), (x_test, y_test) = mnist.load_data()
+
+    return (x_train, y_train), (x_test, y_test)
+
+
+@patch("smdebug.core.config_validator.is_framework_version_supported", return_value=False)
+def test_tensorflow2_with_unsupported_version(eager_mode: bool = True):
+    """ Test the default ZCC behavior of saving losses and metrics in eager and non-eager modes."""
+    smd.del_hook()
+    tf.keras.backend.clear_session()
+    run_eagerly = eager_mode
+    with SagemakerSimulator() as sim:
```

Comment:
Same as above. https://github.com/awslabs/sagemaker-debugger/blob/master/tests/zero_code_change/test_tensorflow2_integration.py#L77

Reply:
The helper function in tf is asserting on hook being created. In the current case, we want to assert on hook not getting. Hence the helper function can not be used here.

Reply:
I think you're looking at a different function (`helper_test_keras_v2`). The function I linked is `helper_keras_fit`, and it doesn't have any assertions (it doesn't even accept a hook as an argument)

Reply:
done